### DISCUSSION TITLE
Operator upgrade - instructions don't mention how to update CRDs

### DISCUSSION THREAD

**Discussion by Andycharalambous** [jaegertracing/jaeger-operator#2096] [Q&A]
Hi,

I want to upgrade from 1.21 to 1.28 but the instructions just say to edit the jaeger-operator deployment (presumably to simply change the image tag to 1.28.0) but no mention if this will auto-magically also update any CRDs that may have changed. Is this how it works or are the upgrade instructions a bit vague?

**Comment by frzifus** [comment]
Hi, could you share the logline/error you face and the crd you try to migrate please?

**Comment by Andycharalambous** [comment]
Hi, sorry if I wasn't clear - the issue is that the instructions for performing an upgrade fail to mention anything about whether CRDs need updating etc. For example, should an upgrade involve downloading the relevant release zip/tar, unpacking it and then kubectl apply-ing the CRD yaml first, before changing the jaeger-operator deployment spec container image tag to the release version?

Thanks

**Reply by joshdcu** [reply]
Hey @Andycharalambous, `Jaeger` CRD updates are outside the scope of the Jaeger operator so yes, you will first have to apply the CRD YAML first before changing the operator image tag in the `Deployment` spec.

One more thing I observed when I updated from `1.29.0` to `1.38.0` is that changes were made to the CRD _without_ a corresponding change in the CRD version (see [this `git diff`](https://github.com/jaegertracing/helm-charts/compare/jaeger-operator-2.27.1...jaeger-operator-2.36.0) and search for `crd.yaml`). This raises a problem because the custom resources based on the CRD cannot take advantage of the new changes to the CRD since the new changes are ignored. We first have to delete the old CRD (note that deleting a CRD also deletes any custom resources based on it!) and then apply the new one.

Hope this helps!
